libraries/typescript/Microsoft.Bot.Builder.Adapters.WeChat/src/attachmentHash.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * @module wechat
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import * as crypto from 'crypto';
+import * as util from 'util';
+
+/**
+ * Attachment hash
+ */
+export class AttachmentHash {
+
+    /**
+     * Calculates the hash value, used to ignore same file when upload media.
+     * @param bytes Bytes content need to be hashed.
+     * @returns  Hash value.
+     */
+    computeBytesHash(bytes: Uint8Array): string {
+        const result = crypto.createHash('md5').update(bytes).digest('hex');
```

Comment:
>md5 [](start = 42, length = 3)

They tend to frown on the use of "md5" in security reviews because it generates collisions. You should use "sha256" instead.

Reply:
> > md5 [](start = 42, length = 3)
> 
> They tend to frown on the use of "md5" in security reviews because it generates collisions. You should use "sha256" instead.

Md5 is used to get a unique hash of the media to avoid uploading the duplicate media, not using it for security.